### PR TITLE
Update and fix create-react-admin

### PR DESCRIPTION
**Pull request by djhi**
## Problem

`create-react-admin` tests fail on `master`

## Solution

- Update `create-react-admin` dependencies
- Fix default template according to default vite template for react
- Fix the tests

## How To Test

In react-admin repository clone directory:
- `make build-create-react-admin install`

In a directory one level above the react-admin repository clone:
- `./node_modules/.bin/create-react-admin myadmin --data-provider ra-data-fakerest --auth-provider local-auth-provider --install npm`
- `cd myadmin`
- `npm run test`

### CODE REVIEW

File: packages/create-react-admin/templates/common/tsconfig.app.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "compilerOptions": {
+    "composite": true,
+    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.app.tsbuildinfo",
+    "target": "ES2020",
+    "useDefineForClassFields": true,
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "module": "ESNext",
+    "skipLibCheck": true,
+
+    /* Bundler mode */
```
Comment (fzaninotto) [inline packages/create-react-admin/templates/common/tsconfig.app.json:11]: this isn't valid JSON

File: packages/create-react-admin/templates/common/package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "private": true,
+    "type": "module",
```
Comment (fzaninotto) [inline packages/create-react-admin/templates/common/package.json:3]: that's a strong choice. If it forces users to use cjs instead of js, I'm -1 for it.

File: packages/create-react-admin/templates/common/vite.config.ts [hunk 3]
```diff
@@ -4,9 +4,6 @@ import react from '@vitejs/plugin-react';
 // https://vitejs.dev/config/
 export default defineConfig({
     plugins: [react()],
-    define: {
```
Comment (fzaninotto) [inline packages/create-react-admin/templates/common/vite.config.ts:7]: Why did you remove it?

File: packages/create-react-admin/templates/ra-data-fakerest/tsconfig.app.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "composite": true,
+    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.app.tsbuildinfo",
+    "target": "ES2020",
+    "useDefineForClassFields": true,
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "module": "ESNext",
+    "skipLibCheck": true,
+    "types": ["vitest/globals"],
+
+    /* Bundler mode */
```
Comment (fzaninotto) [inline packages/create-react-admin/templates/ra-data-fakerest/tsconfig.app.json:12]: same

File: packages/create-react-admin/templates/ra-data-fakerest/vite.config.ts [hunk 5]
```diff
@@ -5,9 +5,6 @@ import react from '@vitejs/plugin-react';
 // https://vitejs.dev/config/
 export default defineConfig({
     plugins: [react()],
-    define: {
```
Comment (fzaninotto) [inline packages/create-react-admin/templates/ra-data-fakerest/vite.config.ts:8]: same

**Review by fzaninotto — CHANGES_REQUESTED** [review]
I'm not sufre I understand how you fixed the test

File: packages/create-react-admin/templates/common/tsconfig.app.json (hunk 1 above)
Comment (djhi) [reply packages/create-react-admin/templates/common/tsconfig.app.json:11]: But it's valid tsconfig though

File: packages/create-react-admin/templates/common/vite.config.ts (hunk 3 above)
Comment (djhi) [reply packages/create-react-admin/templates/common/vite.config.ts:7]: Because it's unnecessary

File: packages/create-react-admin/templates/common/package.json (hunk 2 above)
Comment (djhi) [reply packages/create-react-admin/templates/common/package.json:3]: This is the default for vite templates with TS/React and thanks to the TS config it just work (`moduleResolution=Bundler)

**Review by fzaninotto — APPROVED** [review]
